privcount/traffic_model.py

```diff
@@ -276,8 +281,10 @@ def _get_inter_packet_delays(self, strm_start_ts, byte_events):
             delay = max(long(0), long(micros))
 
             # create a packet delay for each packet
-            while bw_bytes > 0:
+            split_packet_count = 0
+            while bw_bytes > 0 and split_packet_count < TrafficModel.MAX_SPLIT_PACKET_COUNT:
```

Comment:
> I think this means that if a stream sent more than 100 packets "at the same time", those over 100 are ignored and will not be counted by the traffic model as being sent (below, in increment_traffic_counters()). Is that correct?

Yes.

> Won't that bias our model into thinking Tor is sending less than it actually is? Because the traffic model increments counters for i in xrange(num_packets).

Yes, as you say, the traffic model increments counters by generating a list of packets that depends on the bandwidth data it receives. So code uses CPU and RAM proportional to the bandwidth, which could be a huge performance issue for events that are already being sent many times per second. A single large test event in the inject file caused an infinite loop during testing.

That said, connection_bucket_round_robin ensures that each read or write is at most 16 KB (11 packets). So we will never reach the 100 packet limit, and I will add code to log a warning if we do.